### PR TITLE
Add STM32F429 bsp support

### PR DESCRIPTION
**Pull request by grapherd**
Add STM32F429 bsp

    - Test STM32F429 bsp via
      - blinky
      - slinky
      - timtest

### CODE REVIEW

File: hw/bsp/stm32f429discovery/include/bsp/cmsis_nvic.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * CMSIS-style functionality to support dynamic vectors
+ */
+
+#ifndef MBED_CMSIS_NVIC_H
+#define MBED_CMSIS_NVIC_H
+
+#include <stdint.h>
+
+#define NVIC_NUM_VECTORS      (16 + 81)   // CORE + MCU Peripherals
```
Comment (utzig) [inline hw/bsp/stm32f429discovery/include/bsp/cmsis_nvic.h]: The STM32F42x series has (16 + 90) vectors. Please update.

**Comment by mkiiskila** [comment]
There's openocd config scripts present which are not used. I assume these are not meant to stay?
Specifically, f429.cfg & f429discovery.cfg.

**Comment by grapherd** [comment]
@utzig, @mkiiskila since it's merged, should I open a new PR to fixed these issue?

File: hw/bsp/stm32f429discovery/include/bsp/cmsis_nvic.h (hunk 1 above)
Comment (grapherd) [inline hw/bsp/stm32f429discovery/include/bsp/cmsis_nvic.h]: Should this be 91? I found that in st [datasheet](http://www.st.com/content/ccc/resource/technical/document/datasheet/03/b4/b2/36/4c/72/49/29/DM00071990.pdf/files/DM00071990.pdf/jcr:content/translations/en.DM00071990.pdf) page 24 wrote 91:

```
The devices embed a nested vectored interrupt controller able to manage 16 priority levels,
and handle up to 91 maskable interrupt channels plus the 16 interrupt lines of the Cortex®-
M4 with FPU core. 
```

**Pull request merged**